zebra-chain/src/block/height.rs

```diff
@@ -47,19 +50,48 @@ impl Height {
     pub const MAX_AS_U32: u32 = Self::MAX.0;
 }
 
+impl Add<Height> for Height {
+    type Output = Option<Height>;
+
+    fn add(self, rhs: Height) -> Option<Height> {
+        // We know that both values are positive integers. Therefore, the result is
+        // positive, and we can skip the conversions. The checked_add is required,
+        // because the result may overflow.
+        let result = self.0.checked_add(rhs.0)?;
+        match result {
+            h if (Height(h) <= Height::MAX && Height(h) >= Height::MIN) => Some(Height(h)),
+            _ => None,
+        }
```

Comment:
I don't think it makes much sense to use a `match` expression here, we should prefer `if` statements when the match is entirely based on booleans.

```suggestion
        let height = self.0.checked_add(rhs.0)?;
        let height = Height(height);

        if height <= Height::MAX && height >= Height::MIN {
            Some(height)
        } else {
            None
        }
```